db/mongodb.go

```diff
@@ -0,0 +1,337 @@
+		return fmt.Errorf("error creating collection: %w", err)
+	}
+
+	log.Output(1, fmt.Sprintf("Collection %s created successfully", companyTableName))
+
+	if err := m.db.CreateCollection(m.ctx, metaTableName); err != nil {
+		return fmt.Errorf("erro ao criar a coleção: %w", err)
+	}
+
+	log.Output(1, fmt.Sprintf("Collection %s created successfully", metaTableName))
+
+	return nil
+}
+
+// CreateIndexes creates the indexes on the specified collection.
+func (m *MongoDB) CreateIndexes() error {
+
+	fmt.Println("Creating the indexes...")
+
+	collection := m.db.Collection(companyTableName)
+
+	indexes := []mongo.IndexModel{
+		{Keys: bson.D{{Key: "cnpj", Value: 1}}},
+		{Keys: bson.D{{Key: "json.cnpj", Value: 1}}},
+		{Keys: bson.D{{Key: "json.qsa.cpf", Value: 1}}},
+		{Keys: bson.D{{Key: "json.razaosocial", Value: 1}}},
+		{Keys: bson.D{{Key: "json.nomefantasia", Value: 1}}},
+		{Keys: bson.D{{Key: "json.capitalsocial", Value: 1}}},
+		{Keys: bson.D{{Key: "json.telefone1", Value: 1}}},
+		{Keys: bson.D{{Key: "json.telefone2", Value: 1}}},
+		{Keys: bson.D{{Key: "json.naturezajuridica", Value: 1}}},
+		{Keys: bson.D{{Key: "json.qsa.cnpjcpfdosocio", Value: 1}}},
+		{Keys: bson.D{{Key: "json.qsa.qualificaosocio", Value: 1}}},
+		{Keys: bson.D{{Key: "json.qsa.qualificacaorepresentantelegal", Value: 1}}},
+		{Keys: bson.D{{Key: "json.qsa.nomesocio", Value: 1}}},
+		{Keys: bson.D{{Key: "json.bairro", Value: 1}}},
+		{Keys: bson.D{{Key: "json.cep", Value: 1}}},
+		{Keys: bson.D{{Key: "json.cnaefiscal", Value: 1}}},
+		{Keys: bson.D{{Key: "json.cnaessecundarios", Value: 1}}},
+		{Keys: bson.D{{Key: "json.cnaessecundarios.codigo", Value: 1}}},
+		{Keys: bson.D{{Key: "json.datainicioatividade", Value: 1}}},
+		{Keys: bson.D{{Key: "json.email", Value: 1}}},
+		{Keys: bson.D{{Key: "json.codigomunicipio", Value: 1}}},
+		{Keys: bson.D{{Key: "json.codigomunicipioibge", Value: 1}}},
+		{Keys: bson.D{{Key: "json.descricaosituacaocadastral", Value: 1}}},
+		{Keys: bson.D{{Key: "json.uf", Value: 1}}},
+	}
+	_, err := collection.Indexes().CreateMany(m.ctx, indexes)
+	if err != nil {
+		return fmt.Errorf("error creating indexes: %w", err)
+	}
+
+	fmt.Println("Indexes successfully created in the collection:", companyTableName)
+	return nil
+}
+
+// DropCollection exclui completamente uma coleção específica.
+func (m *MongoDB) DropCollection() error {
+
+	collections := []string{companyTableName, metaTableName}
+
+	for _, v := range collections {
+		collection := m.db.Collection(v)
+
+		if err := collection.Drop(m.ctx); err != nil {
+			return fmt.Errorf("error deleting collection: %w", err)
+		}
+
+		fmt.Println("Collection deleted successfully:", v)
+	}
+	m.Close()
+
+	return nil
+}
+
+// CreateCompanies insere uma matriz de dados no MongoDB.
+func (m *MongoDB) CreateCompanies(batch [][]string) error {
+
+	if m == nil {
+		return fmt.Errorf("mongodb connection not initialized")
+	}
+
+	collection := m.db.Collection(companyTableName)
+
+	var empresas []interface{}
```

Comment:
Pq `[]interface{}` aqui? Não seria `[]empresa`?

Reply:
eu deixei interface{} por exigencia da lib do mongo:

https://github.com/mbnunes/minha-receita/blob/mongodb/db/mongodb.go#L159

Reply:
Não entendi a relação da URL com o comentário, pode compartilhar mais sobre essa exigência para eu entender melhor?

Reply:
Afff é na L157 desculpe devo ter clicado errado e copie

Reply:
Ok, `interface{}` em Go quer dizer “qualquer tipo” e, portanto, `[]interface{}` é um _array_ de qualquer tipo de dado.

Se é qualquer tipo de dado, ele deve aceitar `[]empresa`, tendo como base esse exemplo que roda _sem_ erro:

```go
package main

import (
	"fmt"
	"log"
)

func magic(something interface{}) {
	log.Output(1, fmt.Sprintf("Got: %d", something))
}

func main() {
	magic(1)
	magic("one")
}

```

Você tentou com `[]empresa` e deu erro? Qual?

Reply:
![image](https://github.com/user-attachments/assets/228120f4-2db3-45f1-ab47-37f8cb5dac6f)
nao sou tao bom em ingles

Reply:
para completar a informaçao acima:

db/mongodb.go:156:40: cannot use empresas (variable of type []empresa) as []interface{} value in argument to collection.InsertMany

Reply:
Ok, deixe um comentário então, algo como

```go
var empresas []interface{}  // required by MongoDb pkg
```